src/system-core.js

```diff
@@ -242,8 +242,9 @@ var nullContext = Object.freeze(Object.create(null));
 // returns a promise if and only if a top-level await subgraph
 // throws on sync errors
 function postOrderExec (loader, load, seen) {
-  if (seen[load.id])
-    return;
+  if (seen[load.id]) {
```

Comment:
If we have seen it. We should return its execution promise or undefined if it's transitively sync.

Reply:
The problem with this approach, in combination with making `load.E` the promise for the module and its dependencies, is that it will result in deadlock for async cycles. The reason for this return was to notice when there is a cyclical back edge and exit early.

The problem is that this approach is exiting early for edges are not cyclical back references but parallel dependencies.

I think we can fix this by replacing seen with a parentStack that is unique for each `postOrderExec` call, which notes the list of parent modules in the execution. We can then change the check to be `if (parentStack.includes(load.id)) return;` to handle the deadlock cycle back edge.